packages/space-opera/src/components/materials_panel/materials_panel.ts

```diff
@@ -125,12 +136,62 @@ export class MaterialPanel extends ConnectedLitElement {
     this.safeUrlIds = safeUrlIds;
   }
 
+  /* Interpolate from red to original color as n approaches maxN */
+  getInterpolatedColor(original: RGBA, n: number, maxN: number): RGBA {
+    const red: RGBA = [1, 0, 0, 1];
+    const percentRed = (maxN - n) / maxN;
+    const percentOriginal = 1 - percentRed;
+    return [
+      (percentRed * red[0]) + (percentOriginal * original[0]),
+      (percentRed * red[1]) + (percentOriginal * original[1]),
+      (percentRed * red[2]) + (percentOriginal * original[2]),
+      original[3],
+    ];
+  }
+
+  interpolate() {
```

Comment:
logic for interpolating from red to original material, after a new material is selected